### PR TITLE
Add free build flavor without Crashlytics

### LINKED ISSUE #181: Remove Crashlytics?
Personal finances and tracker libraries don't go well together. Would you consider removing Crashlytics, for enhanced privacy? Thanks in advance!

### PR DESCRIPTION
**Pull request by fynngodau**
This pull request adds a `free` build flavor (as opposed to the new `proprietary` build flavor) which does not `compile` the Google Crashlytics dependency. For this, it utilizes the `AnswersProxy` class, which is now different in the different flavors and renamed to `CrashlyticsProxy` to better account for its purpose. In the proprietary one, it is largely unchanged (one minor change to move the code to init Crashlytics¹), in the free one, it doesn't do anything. Closes #181.

¹Not sure how exactly that piece of code works. Functionality shouldn't differ but I haven't tested it.

### CODE REVIEW

**Comment by fynngodau** [comment]
The pull request now also removes dropbox backup/restore feature and thus dependency on `com.dropbox.core:dropbox-core-sdk:3.0.5` from the `free` build flavor.

**Comment by fynngodau** [comment]
Reverting the previous change as it is not necessary unlike I assumed.

File: app/src/free/java/com/blogspot/e_kanivets/moneytracker/util/CrashlyticsProxy.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * In free software builds, this class does nothing as Crashlytics can't be used.
+ * Created on 28/8/19.
+ *
+ * @author Fynn Godau
+ */
+
+public class CrashlyticsProxy {
+    private static CrashlyticsProxy instance;
```
Comment (Yky) [inline app/src/free/java/com/blogspot/e_kanivets/moneytracker/util/CrashlyticsProxy.java]: This singleton implementation is not thread safe. Could be improved with double-checked locking.

File: app/src/proprietary/java/com/blogspot/e_kanivets/moneytracker/util/CrashlyticsProxy.java [hunk 2]
```diff
@@ -1,34 +1,41 @@
...
+ * Debug mode and allow not including Crashlytics in free (fdroid) builds.
  * Created on 1/11/17.
  *
  * @author Evgenii Kanivets
  */
 
-public class AnswersProxy {
-    private static AnswersProxy instance;
+public class CrashlyticsProxy {
+    private static CrashlyticsProxy instance;
```
Comment (Yky) [inline app/src/proprietary/java/com/blogspot/e_kanivets/moneytracker/util/CrashlyticsProxy.java]: see above

**Comment by yevhenii-kanivets** [comment]
@fynngodau looks great (even with a possible thread safety issue pointed out by @Yky). Thanks for your contribution 🙏

**Pull request merged**